tests/test_minimal/test_tools/test_expand_paths.py

```diff
@@ -98,3 +100,47 @@ def test_expand_paths(tmpdir):
     for x in string_directory_out:
         assert x in expected
     assert len(string_directory_out) == len(expected)
+
+
+def test_expand_paths_ibl(tmpdir):
+    expander = LocalPathExpander()
+
+    # set up IBL Steinmetz video file structure
+    generate_path_expander_demo_ibl(tmpdir)
+    base_directory = Path(tmpdir)
+
+    # NOTE: empty brackets used b/c for some reason NR_0021/2022-06-30/ subdir is 002/ instead of 001/
+    out = expander.expand_paths(
+        dict(
+            ibl_video_file=dict(
+                base_directory=base_directory,
+                file_path="steinmetzlab/Subjects/{subject_id}/{session_start_time:%Y-%m-%d}/{}/raw_video_data/_iblrig_leftCamera.raw.{session_id}.mp4",
+            ),
+            ibl_video_directory=dict(
+                base_directory=base_directory,
+                folder_path="steinmetzlab/Subjects/{subject_id}/{session_start_time:%Y-%m-%d}/{}/raw_video_data",
+            ),
+        ),
+    )
+
+    # build expected output from file
+    expected_file_path = Path(__file__).parent / "expand_paths_ibl_expected.json"
+    with open(expected_file_path, "r") as f:
+        expected = json.load(f)
+    for entry in expected:
+        if "NWBFile" in entry["metadata"]:  # `datetime` is not JSON serializable, so convert from string
+            if "session_start_time" in entry["metadata"]["NWBFile"]:
+                datetime_str = entry["metadata"]["NWBFile"].pop("session_start_time")
+                entry["metadata"]["NWBFile"]["session_start_time"] = datetime.strptime(
+                    datetime_str, "%Y-%m-%d %H:%M:%S.%f"
+                )
+        for source_data in entry["source_data"].values():  # update paths with base_directory
+            if "file_path" in source_data.keys():
+                source_data["file_path"] = str(base_directory / source_data["file_path"])
+            if "folder_path" in source_data.keys():
+                source_data["folder_path"] = str(base_directory / source_data["folder_path"])
+
+    # test results
+    for x in out:
+        assert x in expected
+    assert len(out) == len(expected)
```

Comment:
Although, in all honesty you ought to also be able to just do `assert path_expansion_results == expected_results` or otherwise harness `unittestt.TestCase.assertCountEqual` (which has the best error return message of any of the `unittest` variants)